samplers/parser.go

```diff
@@ -51,6 +52,67 @@ func (m *MetricKey) String() string {
 	return buff.String()
 }
 
+// ValidTrace takes in an SSF span and determines if it is valid or not.
+// It also makes sure the Tags is non-nil, since we use it later.
+func ValidTrace(sample ssf.SSFSpan) bool {
+	ret := true
+	ret = ret && sample.Id != 0
+	ret = ret && sample.TraceId != 0
+	ret = ret && sample.StartTimestamp != 0
+	ret = ret && sample.EndTimestamp != 0
+
+	if sample.Tags == nil {
+		sample.Tags = make(map[string]string, 0)
+	}
+
+	return ret
+}
+
+// ValidMetric takes in an SSF sample and determines if it is valid or not.
+func ValidMetric(sample UDPMetric) bool {
+	ret := true
+	ret = ret && sample.Name != ""
+	ret = ret && sample.Value != nil
+	if sample.SampleRate == 0 {
+		sample.SampleRate = 1
+	}
+	return ret
+}
+
+// ErrSSFUnmarshal is returned by ParseSSF when the packet can't be unmarshaled.
+var ErrSSFUnmarshal = errors.New("Unmarshal trace")
```

Comment:
Since you're just using this string as a way to switch on the type of error, then use it in a metric. Could you just return `parse` or `unknown` and use that as the value for the `reason`? That would save you a bunch of code.